Essentials/src/com/earth2me/essentials/Teleport.java

```diff
@@ -123,6 +124,12 @@ protected void now(IUser teleportee, ITarget target, TeleportCause cause) throws
         teleportee.setLastLocation();
         Location loc = target.getLocation();
 
+        UserTeleportEvent event = new UserTeleportEvent(teleportee, cause);
```

Comment:
It might be useful to also pass in the location the player is teleporting to.

Also if the event is cancelled, the teleportee's last location is overridden anyways.